django/utils/dateformat.py

```diff
@@ -202,6 +205,11 @@ def Z(self):
         # Positive offsets have days=0
         return offset.days * 86400 + offset.seconds
 
+    def r(self):
+        # Implement a stub here so that using the `r` format with a date object
+        # raises the correct error.
+        raise NotImplementedError
```

Comment:
This is not strictly related with a bugfix. Can we move this to a separate commit? also test is necessary.

Reply:
I've moved this to a separate commit. The test was done simply by adding `'r'` to the list of formats tested in `test_invalid_time_format_specifiers`. That should be enough, no?